### DISCUSSION TITLE
participant and opponent

### DISCUSSION THREAD

**Discussion by Henry-Lee24** [cyberbotics/wrestling#85] [Q&A] [answered]
I have another question for you. I found that the program I ran when challenging others on the leaderboard was different from the program I ran when others challenged me. I guess it may be due to a color issue, so is the color of my own robot random? In other words, the other robot may be randomly blue or red. Or is it because of other issues? I hope to receive your help and answers, Thank you!

**Comment by omichel** [comment, marked as answer]
During the qualification phase, the color of your robot is red when you are requesting an evaluation, e.g., when committing anything on your repository. It is blue when someone else is requesting an evaluation and they play against your robot at some point.

During the finals with the 32 selected teams, the color will be determined according to the [result](https://github.com/cyberbotics/wrestling/blob/main/result.md) table.

Running games on your own machine may yield to different results than running games on our server machine. Such differences are mainly due to the horsepower of the machine which may be different, the fact that you may run Webots with or without Docker, or on a different operating system, the current load of your machine, the fact the the opponent robot is consuming a lot or little CPU time impacting on the system scheduler, etc. For these reasons, which we can hardly avoid, your controller program should be as robust as possible to handle different speeds of execution.

**Reply by Henry-Lee24** [reply]
I see. Thank you！
